### PR TITLE
refactor(meta): execute revert funcs in blocking manner

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fix https://github.com/risingwavelabs/risingwave/pull/4956#discussion_r963525722.

## Checklist

~~- [ ] I have written necessary rustdoc comments~~
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/meta/src/stream/scale.rs [hunk 1]
```diff
@@ -219,19 +220,25 @@ where
...
-                    }
-                });
-            },
-        );
+        let mut revert_funcs = vec![];
+        if let Err(e) = self
+            .reschedule_actors_impl(&mut revert_funcs, reschedule)
+            .await
+        {
+            for revert_func in revert_funcs {
```
Comment (BugenZhao) [inline src/meta/src/stream/scale.rs:228]: Do we need to reverse the order?

Comment (zwang28) [reply src/meta/src/stream/scale.rs]: Yes, fixed.

**Comment by zwang28** [comment]
Currently revert_funcs are required to be either infallible or can handle err locally itself.

If we allow revert_funcs to be fallible, we need decide the behavior of revert_funcs execution when one of them returns Err.

**Review by BugenZhao — APPROVED** [review]
LGTM! Looks like there's no better way to simulate async drop in Rust for now. This change makes it much clearer at least.

**Pull request merged**